Failing to build Watson: MonadFail not in scope

Tried to install holmes from Cabal and got this error:
```
src/Control/Monad/Watson.hs:55:10: error:
    Not in scope: type constructor or class ‘MonadFail’
   |
55 | instance MonadFail (Watson h) where
   |          ^^^^^^^^^
```

I'm using GHC version 8.6.5, it seems to be failing on Hackage CI as well.

--- a/src/Control/Monad/Holmes.hs
+++ b/src/Control/Monad/Holmes.hs
@@ -36,6 +36,7 @@ module Control.Monad.Holmes
   , whenever
   ) where
 
+import Control.Monad.Fail (MonadFail, fail)
 import Control.Monad.Cell.Class (MonadCell (..))
 import Control.Monad.IO.Class (MonadIO (..))
 import qualified Control.Monad.Cell.Class as Cell

--- a/src/Control/Monad/Holmes.hs
+++ b/src/Control/Monad/Holmes.hs
@@ -6,6 +6,7 @@
 {-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE CPP #-}
 
 {-|
 Module      : Control.Monad.Holmes
@@ -36,7 +37,9 @@ module Control.Monad.Holmes
   , whenever
   ) where
 
+#if __GLASGOW_HASKELL__ == 806
 import Control.Monad.Fail (MonadFail, fail)
+#endif
 import Control.Monad.Cell.Class (MonadCell (..))
 import Control.Monad.IO.Class (MonadIO (..))
 import qualified Control.Monad.Cell.Class as Cell

--- a/src/Control/Monad/Watson.hs
+++ b/src/Control/Monad/Watson.hs
@@ -30,6 +30,7 @@ module Control.Monad.Watson
   , whenever
   ) where
 
+import Control.Monad.Fail (MonadFail, fail)
 import Control.Monad.ST (ST, runST)
 import Control.Monad.Cell.Class (MonadCell (..))
 import qualified Control.Monad.Cell.Class as Cell

--- a/src/Control/Monad/Watson.hs
+++ b/src/Control/Monad/Watson.hs
@@ -4,6 +4,7 @@
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE CPP #-}
 
 {-|
 Module      : Control.Monad.Watson
@@ -30,7 +31,9 @@ module Control.Monad.Watson
   , whenever
   ) where
 
+#if __GLASGOW_HASKELL__ == 806
 import Control.Monad.Fail (MonadFail, fail)
+#endif
 import Control.Monad.ST (ST, runST)
 import Control.Monad.Cell.Class (MonadCell (..))
 import qualified Control.Monad.Cell.Class as Cell

